### PR TITLE
Validate context if there is any of the requested space types available

### PR DESCRIPTION
**Pull request by MatthiasThDs**
I put this function into common:

```
def is_valid_space(context, allowed_spaces):
    for area in context.screen.areas:
        for space in area.spaces:
            if space.type in allowed_spaces:
                return True
    return False
```

And use it in all the operators.

### CODE REVIEW

File: src/magic_uv/op/align_uv_cursor.py [hunk 1]
```diff
@@ -33,20 +33,15 @@
...
-        if (space.type == 'IMAGE_EDITOR') or (space.type == 'VIEW_3D'):
-            break
-    else:
+    # after the execution    
+    if not common.is_valid_space(context, ['IMAGE_EDITOR', 'VIEW_3D']):
         return False
 
     return True
 
-
```
Comment (nutti) [inline src/magic_uv/op/align_uv_cursor.py:49]: Please do not delete this line?

File: src/magic_uv/op/align_uv_cursor.py [hunk 2]
```diff
@@ -33,20 +33,15 @@
 from ..utils.property_class_registry import PropertyClassRegistry
 from ..utils import compatibility as compat
 
-
```
Comment (nutti) [inline src/magic_uv/op/align_uv_cursor.py:36]: Please do not delete this line?

File: src/magic_uv/op/align_uv_cursor.py [hunk 3]
```diff
@@ -33,20 +33,15 @@
...
 def _is_valid_context(context):
     # 'IMAGE_EDITOR' and 'VIEW_3D' space is allowed to execute.
     # If 'View_3D' space is not allowed, you can't find option in Tool-Shelf
-    # after the execution
```
Comment (nutti) [inline src/magic_uv/op/align_uv_cursor.py:40]: I think this change makes the CI fail.
Could you delete the spaces following this comment?

File: src/magic_uv/op/copy_paste_uv.py [hunk 4]
```diff
@@ -49,10 +49,7 @@ def _is_valid_context(context):
         return False
 
     # only 'VIEW_3D' space is allowed to execute
-    for space in context.area.spaces:
-        if space.type == 'VIEW_3D':
-            break
-    else:
+    if not common.is_valid_space(context, ['IMAGE_EDITOR', 'VIEW_3D']):
```
Comment (nutti) [inline src/magic_uv/op/copy_paste_uv.py:52]: I think this operation should be executed only in View3D.

```suggestion
    if not common.is_valid_space(context, ['VIEW_3D']):
```

**Review by nutti — CHANGES_REQUESTED** [review]
Thanks for your great contribution and improvements.
Some CI tests are failed because of the invalid code format.

Could you fix them?

**Comment by nutti** [comment]
@MatthiasThDs 

BTW, I will add your name to [here](https://github.com/nutti/Magic-UV#project-authors) and [here](https://github.com/nutti/Magic-UV/blob/master/src/magic_uv/__init__.py#L29-L31) after this PR is merged.
Do you have any requests about your name?

If you don't have any requests, I will add @MatthiasThDs (MatthiasThDs) to there.

**Comment by MatthiasThDs** [comment]
Hi,
yes MatthiasThDs is absolutely ok.

File: src/magic_uv/op/align_uv_cursor.py [hunk 5]
```diff
@@ -33,20 +33,15 @@
...
-    for space in context.area.spaces:
-        if (space.type == 'IMAGE_EDITOR') or (space.type == 'VIEW_3D'):
-            break
-    else:
+    # after the execution
```
Comment (nutti) [inline src/magic_uv/op/align_uv_cursor.py:39]:
```suggestion
    # after the execution
```

File: src/magic_uv/op/align_uv_cursor.py [hunk 6]
```diff
@@ -33,20 +33,15 @@
 from ..utils.property_class_registry import PropertyClassRegistry
 from ..utils import compatibility as compat
```
Comment (nutti) [inline src/magic_uv/op/align_uv_cursor.py:35]:
```suggestion


```

Comment (nutti) [inline src/magic_uv/op/align_uv_cursor.py:35]:
```suggestion

```

Comment (nutti) [inline src/magic_uv/op/align_uv_cursor.py]:
```suggestion


```

Comment (nutti) [inline src/magic_uv/op/align_uv_cursor.py]:
```suggestion


```

File: src/magic_uv/op/align_uv_cursor.py [hunk 7]
```diff
@@ -33,20 +33,15 @@
...
-    # after the execution
-    for space in context.area.spaces:
-        if (space.type == 'IMAGE_EDITOR') or (space.type == 'VIEW_3D'):
-            break
-    else:
+    # after the execution    
+    if not common.is_valid_space(context, ['IMAGE_EDITOR', 'VIEW_3D']):
         return False
 
     return True
```
Comment (nutti) [inline src/magic_uv/op/align_uv_cursor.py:43]:
```suggestion


```

**Review by nutti — CHANGES_REQUESTED** [review]
@MatthiasThDs 

Thanks.
Could you apply these suggestion?
I could not merge your PR if these suggestions are not approved.

File: src/magic_uv/op/align_uv_cursor.py [hunk 8]
```diff
@@ -32,21 +32,15 @@
 from ..utils.bl_class_registry import BlClassRegistry
 from ..utils.property_class_registry import PropertyClassRegistry
 from ..utils import compatibility as compat
```
Comment (nutti) [inline src/magic_uv/op/align_uv_cursor.py:34]:
```suggestion
from ..utils import compatibility as compat


```

File: src/magic_uv/op/align_uv_cursor.py [hunk 9]
```diff
@@ -32,21 +32,15 @@
...
     # If 'View_3D' space is not allowed, you can't find option in Tool-Shelf
     # after the execution
-    for space in context.area.spaces:
-        if (space.type == 'IMAGE_EDITOR') or (space.type == 'VIEW_3D'):
-            break
-    else:
+    if not common.is_valid_space(context, ['IMAGE_EDITOR', 'VIEW_3D']):
         return False
 
     return True
```
Comment (nutti) [inline src/magic_uv/op/align_uv_cursor.py]:
```suggestion


```

File: src/magic_uv/common.py [hunk 10]
```diff
@@ -43,6 +43,12 @@ def is_console_mode():
         return False
     return os.environ["MUV_CONSOLE_MODE"] == "true"
```
Comment (nutti) [inline src/magic_uv/common.py:45]:
```suggestion


```

File: src/magic_uv/common.py [hunk 11]
```diff
@@ -43,6 +43,12 @@ def is_console_mode():
...
 
+def is_valid_space(context, allowed_spaces):
+    for area in context.screen.areas:
+        for space in area.spaces:
+            if space.type in allowed_spaces:
+                return True
+    return False
```
Comment (nutti) [inline src/magic_uv/common.py:52]:
```suggestion


```

**Review by nutti — CHANGES_REQUESTED** [review]
@MatthiasThDs 

Thanks.
Sorry, the CI still error because of the lack of blank line.
Could you apply these suggestions?

**Comment by MatthiasThDs** [comment]
I hope we could make it.
Sry for having the wrong white spaces.
Thank you very much for your support on this issue!

**Review by nutti — APPROVED** [review]
LGTM, thanks for your great contribution!

**Comment by nutti** [comment]
@MatthiasThDs 

Merged this PR. Thanks for great work!

**Pull request merged**